### PR TITLE
Add.container.theme

### PR DESCRIPTION
**Pull request by caseyflynn-google**
Review for current work.

### CODE REVIEW

File: bundles/org.eclipse.orion.client.ui/web/defaults.pref~ [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+	"/plugins":{
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/defaults.pref~]: I'd remove this file as it looks like its an autosaved file.

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/ThemeVersion.js [hunk 2]
```diff
@@ -7,14 +7,15 @@
...
+ * 				 Casey Flynn - Google Inc.
  ******************************************************************************/
 /*eslint-env browser, amd*/
 define([], function() {
 
 	/**
 	 * Version string for theme data. Please update this string whenever you change the style of a themable element.
 	 */
-	var THEMES_VERSION = "6.88";
+	var THEMES_VERSION = "6.7";
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/ThemeVersion.js]: Wouldn't you want to increase the themes version?

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeData.js [hunk 3]
```diff
@@ -217,88 +863,16 @@ define([
 		function getThemeStorageInfo(){
 			return {
 				storage:'/themes',
-				styleset:'styles',
-				defaultTheme:'Orion2014',
-				selectedKey: 'selected',
+				styleset:'containerStyles',
+				defaultTheme:'lightPage',
+				//defaultTheme:'orionPage',
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeData.js]: I assume that when you send this for the real review, you'll remove out all the commented code?

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeData.js [hunk 4]
```diff
@@ -217,88 +863,16 @@ define([
...
-				selectedKey: 'selected',
+				styleset:'containerStyles',
+				defaultTheme:'lightPage',
+				//defaultTheme:'orionPage',
+				selectedKey: 'containerTheme',
 				version: THEMES_VERSION
 			};
 		}
-
-		ThemeData.prototype.getThemeStorageInfo = getThemeStorageInfo;
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeData.js]: Whats this deleted section? Dead code?

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeSheetWriter.js [hunk 5]
```diff
@@ -7,381 +7,51 @@
...
  ******************************************************************************/
 /*eslint-env browser, amd*/
 
-define(['orion/widgets/themes/ThemeClass'], 
+define(['orion/widgets/themes/ThemeClass',
+],
 	function(ThemeClass) {
-		
-		// These values are used to override various themeable element -> color mappings provided by theme data.
-		var navbar = 'white';
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeSheetWriter.js]: Is this also all dead code?

File: bundles/org.eclipse.orion.client.ui/web/defaults.pref~ (hunk 1 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/defaults.pref~]: Sorry, I included too many files in the commit I will remove this, and the ThemeVersion.

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/ThemeVersion.js (hunk 2 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/ThemeVersion.js]: This shouldn't have been included in this commit, as we will want to have the Container theme builder completed before the styling takes effect.

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeSheetWriter.js (hunk 5 above)
Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeSheetWriter.js]: Yes. This code is never called, and has been replaced by the new method of generating CSS.

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeData.js [hunk 6]
```diff
@@ -217,88 +863,15 @@ define([
 		function getThemeStorageInfo(){
 			return {
 				storage:'/themes',
-				styleset:'styles',
-				defaultTheme:'Orion2014',
-				selectedKey: 'selected',
+				styleset:'containerStyles',
+				defaultTheme:'lightPage',
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeData.js]: SHould the default be orion so we don't have any visible changes?

File: bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeSheetWriter.js [hunk 7]
```diff
@@ -7,381 +7,51 @@
...
-			var navlink = new ThemeClass.ThemeClass( 'navlink' );
-			navlink.style.display = 'inline-block';
-			navlink.style.padding = '2px';
-			navlink.style.color = this.content;
-			navlink.style.verticalAlign = 'bottom';
-			
-			styles.push( navlink );
 
+		function prototypeStyleWriter(settings) {
+			var parseStyles = function(object, ancestors, className, result) {
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/orion/widgets/themes/container/ThemeSheetWriter.js]: how about a better name then object?

File: bundles/org.eclipse.orion.client.ui/web/js-tests/ui/widgets/widgetsTests.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+				var expected = "\n"+
+					".orionPage .selector {\n"+
+					"\tattribute: value;\n"+
+					"}";
+				
+				
+				var actual = themeSheetWriter.getSheet("orionPage", settings);
+				assert.equal(actual, expected);				
+			});
+			it("Should return an empty style if no className is specified", function() {
```
Comment (wilford) [inline bundles/org.eclipse.orion.client.ui/web/js-tests/ui/widgets/widgetsTests.js]: Just for completeness, lets add some more complex test cases:  Multiple inheritance, a ancestor having mutlitple children, multipl siblings, etc.

Comment (caseyflynn-google) [inline bundles/org.eclipse.orion.client.ui/web/js-tests/ui/widgets/widgetsTests.js]: Sounds good.

**Comment by caseyflynn-google** [comment]
PTAL

**Pull request closed without merging**